support-frontend/assets/pages/supporter-plus-landing/components/paymentTsAndCs.tsx

```diff
@@ -145,6 +149,20 @@ export function PaymentTsAndCs({
 		);
 	}
 
+	const studentPricePeriod = `${studentDiscount?.discountPriceWithCurrency} per ${studentDiscount?.periodNoun}`;
```

Comment:
Since `studentDiscount` is possible undefined, is there a danger that `studentPricePeriod` could be the string `" per "`?

Reply:
Its possible if this studentPricePeriod was used without a check, have refactored it into a function now to ensure it cannot happen. Updated.